src/main/java/net/smartcosmos/dao/metadata/impl/MetadataPersistenceService.java

```diff
@@ -52,42 +39,111 @@ public MetadataPersistenceService(MetadataRepository metadataRepository,
     }
 
     @Override
-    public List<MetadataResponse> upsert(String accountUrn, Collection<MetadataUpsert> upsertMetadataCollection) throws ConstraintViolationException {
+    public Optional<MetadataResponse> create(String tenantUrn, MetadataCreate createMetadata)
+        throws ConstraintViolationException {
+
+        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
```

Comment:
This can throw `IllegalArgumentException`, please catch the exception and write log output consistent with `ThingPersistenceService`.